Makefile

```diff
@@ -41,10 +41,21 @@ endif
 
 # PAM support will only be built into Teleport if headers exist at build time.
 PAM_MESSAGE := "without PAM support"
+ifneq ("$(OS)","darwin")
 ifneq ("$(wildcard /usr/include/security/pam_appl.h)","")
 PAM_TAG := pam
 PAM_MESSAGE := "with PAM support"
 endif
+endif
+
+# PAM headers for Darwin live under /usr/local/include/security instead, as SIP
+# prevents us from modifying/creating /usr/include/security on newer versions of MacOS
+ifeq ("$(OS)","darwin")
+ifneq ("$(wildcard /usr/local/include/security/pam_appl.h)","")
+PAM_TAG := pam
+PAM_MESSAGE := "with PAM support"
+endif
+endif
```

Comment:
Why not expand the `ifeq` statements above instead of making this another block?

Reply:
Maybe I'm being dense, but what do you mean by this? I think I could rewrite it this way to drop the Darwin check (as it looks like cgo includes `/usr/include` and `/usr/local/include` by default):

```make
# PAM support will only be built into Teleport if headers exist at build time.
PAM_MESSAGE := "without PAM support"
ifneq ("$(wildcard /usr/include/security/pam_appl.h)","")
PAM_TAG := pam
PAM_MESSAGE := "with PAM support"
else
# PAM headers for Darwin live under /usr/local/include/security instead, as SIP
# prevents us from modifying/creating /usr/include/security on newer versions of MacOS
ifneq ("$(wildcard /usr/local/include/security/pam_appl.h)","")
PAM_TAG := pam
PAM_MESSAGE := "with PAM support"
endif
endif
```

I'm not sure that's any much clearer/simpler though.

Reply:
(I'm pretty sure `ifeq` and `ifneq` only support one condition for a test)

Reply:
Okay, makes sense.